### PR TITLE
Add lambda to replace status JSON in the HUD

### PR DESCRIPTION
**Pull request by driazati**
The JSON is often out of date or missing (pop open the network tab to see many 404 errors), this replaces it with a direct DB query.

### CODE REVIEW

File: aws/lambda/hud-query-proxy/lambda_function.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+import asyncio
+import aiomysql
+
+
+def eprint(*args):
+    print(*args, file=sys.stderr)
+
+
+async def run_query(query: str, params: List[str] = None) -> Dict[str, Any]:
+    eprint(f"Executing '{query}' with params: {params}")
```
Comment (malfet) [inline aws/lambda/hud-query-proxy/lambda_function.py:31]: Isn't there a better way to write something to the log?

File: aws/lambda/hud-query-proxy/lambda_function.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+    my_con = await aiomysql.connect(
+        host=os.environ["db_host"],
+        port=3306,
```
Comment (malfet) [inline aws/lambda/hud-query-proxy/lambda_function.py:35]: Nit
```suggestion
        port=socket.getservbyname("mysql"),
```

File: aws/lambda/hud-query-proxy/lambda_function.py [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+import io
+import boto3
+
+from typing import *
+
+import asyncio
+import aiomysql
+
+
+def eprint(*args):
```
Comment (malfet) [inline aws/lambda/hud-query-proxy/lambda_function.py:26]: Nit
```suggestion
def eprint(*args) -> None:
```

File: aws/lambda/hud-query-proxy/lambda_function.py [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+            head_commit_author_username,
+            head_commit_message
+        from push_event
+        where ref = %s
+        order by head_commit_timestamp desc
+        limit %s;
+        """,
+        (branch, NUM_COMMITS),
+    )
+    for row in results:
```
Comment (malfet) [inline aws/lambda/hud-query-proxy/lambda_function.py:65]: Hmm, if `run_query` indeed returns `Dict[str, Any]`, then row is `str`, isn't i?

File: aws/lambda/hud-query-proxy/lambda_function.py [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+            pr = "<unknown>"
+
+        row["head_commit_title"] = title
+        row["pr"] = pr
+        row["jobs"] = []
+
+    return results
+
+
+async def gha_jobs_for_shas(shas: List[str]):
```
Comment (malfet) [inline aws/lambda/hud-query-proxy/lambda_function.py:87]: Please add type annotation for return type

**Review by malfet — APPROVED** [review]
Looks good to me, although can you please clarify in lambda comment how/when it supposed to be executed?
And can you please explain why querying DB is preferable to creating the same data using statuses already in S3?

**Review by seemethere — APPROVED** [review]

**Comment by driazati** [comment]
> Looks good to me, although can you please clarify in lambda comment how/when it supposed to be executed? And can you please explain why querying DB is preferable to creating the same data using statuses already in S3?

It's nice since we're collecting this info anyways, we can have a single pipeline that we monitor (and the lambda to sync to the db has been more reliable than the status webhook lately). We certainly could debug that, but this approach seems like less code overall to maintain and easier to track. Also the data is not all in S3 since some commits are missing and others have outdated statuses

File: aws/lambda/hud-query-proxy/lambda_function.py (hunk 1 above)
Comment (driazati) [reply aws/lambda/hud-query-proxy/lambda_function.py:31]: sure we could use `logging` but it ends up the same thing anyways, cloudwatch auto-adds in things like timestamps so printing them here is redundant

**Pull request merged**